src/apify_client/clients/base/actor_job_base_client.py

```diff
@@ -1,13 +1,40 @@
-from typing import Optional
+import math
+from datetime import datetime
+from typing import Dict, Optional
 
+from ..._consts import TERMINAL_ACTOR_JOB_STATUSES, ActorJobStatus
+from ..._utils import _parse_date_fields, _pluck_data
 from .resource_client import ResourceClient
 
-# from ..._utils import _parse_date_fields, _pluck_data
-
 
 class ActorJobBaseClient(ResourceClient):
     """Base sub-client class for actor runs and actor builds."""
 
-    def _wait_for_finish(self, wait_secs: Optional[int] = None) -> None:
-        # TODO implement when implementing run and build endpoints
-        pass
+    def _wait_for_finish(self, wait_secs: Optional[int] = None) -> Dict:
+        started_at = datetime.now()
+        should_repeat = True
+        job: Optional[Dict] = None
+        seconds_elapsed = 0
+
+        while(should_repeat):
+            wait_for_finish = 9999
```

Comment:
We could have the `wait_secs` parameter have 9999 as a default value, and not have it optional - that way you can simplify the logic here, and it will be explicitly obvious to the caller how long will this function wait by default. Also, let's use 999999 just as the JS client, sometimes you might want to wait for more than ~3 hours for the run to finish.

Reply:
It'd make the logic simpler, but I think that it's cleaner to pass `None` when user means "I don't care about wait secs. I just want to wait indefinitely". So I'm for keeping optional here. Changing default to 999999

Reply:
Agreed, it makes sense now that internally we are waiting 999999 seconds, which is basically infinitely.